token-syncer/src/token_syncer/commands/syncer.clj

```diff
@@ -271,28 +275,46 @@
       (log/error th "unable to sync tokens")
       (throw th))))
 
+(defn valid-metadata-name?
+  "Validates that a non-nil input passes as a vlaid metadata name:
+   - contains fewer than 100 characters; and
+   - made up of lower-case letters, numbers, and hyphens and must start with a letter."
+  [metadata-name]
+  (or (nil? metadata-name)
+      (and (< (count metadata-name) 100)
+           (some? (re-matches #"^[a-z][a-z0-9\\-]*$" metadata-name)))))
+
 (def sync-clusters-config
   {:execute-command (fn execute-sync-clusters-command
                       [{:keys [waiter-api]} {:keys [options]} arguments]
-                      (let [{:keys [limit]} options
+                      (let [{:keys [limit metadata-name]} options
                             cluster-urls-set (set arguments)]
                         (cond
                           (<= (-> cluster-urls-set set count) 1)
                           {:exit-code 1
                            :message (str "at least two different cluster urls required, provided: " (vec arguments))}
 
                           :else
-                          (let [sync-result (sync-tokens waiter-api cluster-urls-set limit)
+                          (let [{:keys [details summary] :as sync-result} (sync-tokens waiter-api cluster-urls-set limit metadata-name)
                                 exit-code (-> (get-in sync-result [:summary :sync :failed])
-                                              empty?
-                                              (if 0 1))]
-                            (log/info (-> sync-result pp/pprint with-out-str str/trim))
+                                            empty?
+                                            (if 0 1))]
+                            (log/info "sync details:" (into (sorted-map) details))
+                            (log/info "sync summary:" (into (sorted-map) summary))
```

Comment:
Note: Unrelated change to this PR to reduce number if lines output by the syncer.